DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs

```diff
@@ -591,6 +594,145 @@ private void PostUIInit()
+                bool hasSuperWeapons = Conversions.BooleanFromString(splitMessage[12], false);
+                bool hasCrates = Conversions.BooleanFromString(splitMessage[13], false);
+
+                CnCNetGame cncnetGame = gameCollection.GetGameFromHostedChannelName(e.ChannelName);
+                CnCNetTunnel tunnel = tunnelHandler.Tunnels.Find(t => t.Address == tunnelAddress && t.Port == tunnelPort);
+
+                if (tunnel == null || cncnetGame == null)
+                {
+                    Logger.Log($"GameBroadcastChannel_ChannelListReceived ** Tunnel: {tunnel} // cncnetGame: {cncnetGame}");
+                    return;
+                }
+
+                HostedCnCNetGame game = new HostedCnCNetGame(
+                    gameRoomChannelName,
+                    revision,
+                    gameVersion,
+                    maxPlayers,
+                    gameRoomDisplayName,
+                    isCustomPassword,
+                    true,
+                    players,
+                    hostName,
+                    mapName,
+                    gameMode
+                );
+
+                game.IsLoadedGame = isLoadedGame;
+                game.MatchID = loadedGameId;
+                game.LastRefreshTime = DateTime.Now;
+                game.IsLadder = isLadder;
+                game.Game = cncnetGame;
+                game.Locked = locked || (game.IsLoadedGame && !game.Players.Contains(ProgramConstants.PLAYERNAME));
+                game.Incompatible = cncnetGame == localGame && game.GameVersion != ProgramConstants.GAME_VERSION;
+                game.TunnelServer = tunnel;
+                game.HasSpecialGameMode = hasSpecialGameMode;
+                game.HasCrates = hasCrates;
+                game.HasSuperWeapons = hasSuperWeapons;
+
+                // If the game is closed, remove it from the list
+                if (isClosed)
+                {
+                    int index = lbGameList.HostedGames.FindIndex(hg => ((HostedCnCNetGame)hg).HostName == hostName);
+
+                    if (index > -1)
+                    {
+                        lbGameList.RemoveGame(index);
+
+                        // Dismiss any outstanding invitations that are no longer valid
+                        DismissInvalidInvitations();
+                    }
+
+                    return;
+                }
+
+                int gameIndex = lbGameList.HostedGames.FindIndex(hg => ((HostedCnCNetGame)hg).HostName == hostName);
+                if (gameIndex > -1)
+                {
+                    lbGameList.HostedGames[gameIndex] = game;
+                }
+                else
+                {
+                    if (UserINISettings.Instance.PlaySoundOnGameHosted &&
+                        cncnetGame.InternalName == localGameID.ToLower() &&
+                        !ProgramConstants.IsInGame && !game.Locked)
+                    {
+                        SoundPlayer.Play(sndGameCreated);
+                    }
```

Comment:
I am pretty sure the initial fetch with 100 games is gonna destroy your eardrums if the new game sound is on. Also since you're touching this code anyways - for the sake of better readability we now place operators at the beginning of a split line, not at the end.